### PR TITLE
feat: add Chatbase class

### PR DESCRIPTION
**Pull request by tonyghiani**
- get method to retrieve messages
- put method to insert a message

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+const { Pool } = require('pg');
+
+class Chatbase {
+  constructor() {
+    this.pool = new Pool({
+      user: process.env.user || argValidation('user'),
```
Comment (nandanrao) [inline lib/index.js]: Why lowercase env vars?? I would always do uppercase, more canonical. Also, why not use ESNext default arguments if you want what are essentially optional arguments?

**Review by nandanrao — COMMENTED** [review]
Other than that looks good, let's see some tests!

File: lib/index.js (hunk 1 above)
Comment (tonyghiani) [reply lib/index.js]: Ops distraction error, my bad! I always follow the upper case convention, i'll switch to default argument!

File: lib/index.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  async get (key) {
+    const result = await this.pool.query('SELECT * FROM messages WHERE u_key = $1 ORDER BY timestamp ASC', [key]);
+    return result.rows;
+  }
+
+  // Put method to insert a new message related to a single user
+  async put (key, message) {
+    let exists = { rows: [true]};
+    let hash = '';
+    // Generate a new hash in case an existing one is found as unique id in the database
```
Comment (nandanrao) [inline lib/index.js]: I would push this work to the DB. No point in making multiple requests. Just create the hash and insert it as a column (so 3 columns) -- and set a UNIQUE constraint on the id column (or make it the primary key)

File: lib/chatbase.test.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+const { Pool } = require('pg');
+const mocha = require('mocha')
+const chai = require('chai')
+const should = chai.should()
+const Chatbase = require('./index')
+
+describe('Chatbase Postgres', () => {
+  const mockMessage = {
```
Comment (nandanrao) [inline lib/chatbase.test.js]: Where did you get this format?

Comment (nandanrao) [reply lib/chatbase.test.js]: The format is put(key, value) where the value is the JSON data itself. Therefore "message" in your case will be the string data, and this client will be responsible for adding the timestamp.

Comment (nandanrao) [reply lib/chatbase.test.js]: Maybe for now add timestamp as an optional parameter with default Date.now()  -- 

```
put(key, value, timestamp = Date.now())
```

I'm thinking we should actually store timestamp outself on reception in Kafka...

Comment (nandanrao) [reply lib/chatbase.test.js]: I take it back. Let's change the format. Make timestamp required: 

```
put(key, value, timestamp)
```

File: lib/index.js (hunk 2 above)
Comment (tonyghiani) [reply lib/index.js]: Yes i was considering to use it as primary key!

**Pull request merged**